Include instructions checking TLS routing is on or not

stevenGravy opened the pull request:
- provides a hidden detail on how to confirm if TLS routing is on or not
- Shows tsh proxy db tunneling as the default for cloud.

stevenGravy committed:
Add instructions on how to tell if TLS routing is enabled and use tunnelling example for cloud

alexfornuto commented:
@stevenGravy would you mind if I took a pass on the formatting here? Instead of creating the partial file `tsh-db-tunnel.mdx` to use twice in the same doc, I think we can adjust the tabs using the dropdown option to create a more cohesive adjustable block for all use cases.

ptgott commented on docs/pages/includes/determinetls.mdx:
```diff
@@ -0,0 +1,26 @@
+<Details title="How to tell if TLS Routing is Enabled" opened={false}>
```
I think we can remove the `Details` wrapper, since it could be helpful to confirm whether TLS Routing is enabled or not in all cases, and doesn't add much time.

ptgott commented on docs/pages/includes/determinetls.mdx:
```diff
@@ -0,0 +1,26 @@
+<Details title="How to tell if TLS Routing is Enabled" opened={false}>
+
+<ScopedBlock scope={["oss", "enterprise"]}>
```
Is the only difference between the `ScopedBlock`s the address of the cluster? We might consider using a `Var` for the address and instructing the user to fill it in.

alexfornuto replied:
I think a Var is overkill for single use. I'm going to consolidate with a note to change the address as appropriate.

ptgott commented on docs/pages/includes/determinetls.mdx:
```diff
@@ -0,0 +1,26 @@
+<Details title="How to tell if TLS Routing is Enabled" opened={false}>
+
```
To prevent the user from leaving the page, I think it would be useful to add a one-sentence definition of TLS Routing here. This way, we can remove the link to the TLS Routing page later in these instructions.

stevenGravy replied:
addressed

ptgott commented on docs/pages/connect-your-client/gui-clients.mdx:
```diff
@@ -116,7 +100,10 @@ your GUI client below. When entering the hostname, use `localhost` rather than
+
+(!docs/pages/includes/determinetls.mdx!)
+
+If you're not using [TLS routing](../management/operations/tls-routing.mdx), run the following command to see the database
```
```suggestion
If you're not using [TLS Routing](../management/operations/tls-routing.mdx), run the following command to see the database
```

ptgott replied:
Also, as mentioned in another comment, I think it would be  smoother to introduce TLS Routing at the top of these instructions in one sentence and remove this link. This way, the user wouldn't need to leave the page.

ptgott commented on docs/pages/includes/determinetls.mdx:
```diff
@@ -0,0 +1,26 @@
+
+<ScopedBlock scope={["oss", "enterprise"]}>
+To get details of the Teleport Cluster you can invoke `/webapi/ping` path
+on the proxy service.  The `proxy.tls_routing_enabled` value provides if TLS routing is enabled.
```
```suggestion
on the Teleport Proxy Service. The `proxy.tls_routing_enabled` value indicates if TLS Routing is enabled.
```

alexfornuto replied:
recreating in my edits since I'm making other changes to this line.

ptgott commented on docs/pages/includes/determinetls.mdx:
```diff
@@ -0,0 +1,26 @@
+<ScopedBlock scope={["oss", "enterprise"]}>
+To get details of the Teleport Cluster you can invoke `/webapi/ping` path
+on the proxy service.  The `proxy.tls_routing_enabled` value provides if TLS routing is enabled.
+Here's an example for the proxy on `https://teleport.example.com`:
```
```suggestion
Here's an example for the Proxy Service instance on `https://teleport.example.com`:
```

Commit:
copy edits

alexfornuto commented:
I addressed all of the copy-related suggestions from @ptgott's review, turning this back to @stevenGravy to address the remaining concerns.

stevenGravy committed:
verbiage update

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

stevenGravy committed:
Merge branch 'master' into stevenGravy/docs/tlsrouting

stevenGravy committed:
Provide definition on TLS routing

stevenGravy commented:
> I addressed all of the copy-related suggestions from @ptgott's review, turning this back to @stevenGravy to address the remaining concerns.

Addressed comments on including brief TLS description and removed TLS routing link.

ptgott commented on docs/pages/connect-your-client/gui-clients.mdx:
```diff
@@ -71,34 +71,16 @@ Ensure that your environment includes the following:
 
```
I know this wasn't part of the PR, but at the top of the "Get connection information" section, I'm not sure a reader will be able to determine which `TabItem` to select. Can we include a quick introduction to what "Authenticated Proxy", "TLS Routing", and "Separate Ports" mean, and when a reader should select one of these tabs?

ptgott replied:
@stevenGravy What do you think about the comment above?

ptgott commented on docs/pages/connect-your-client/gui-clients.mdx:
```diff
@@ -71,34 +71,16 @@ Ensure that your environment includes the following:
-the following command to start a local TLS proxy your GUI database client will
-be connecting to:
+
+(!docs/pages/includes/determinetls.mdx!)
```
Should we include the instructions to determine whether TLS Routing is enabled *before* determining which approach to use to connect to the database, i.e., the authenticated proxy, `tsh proxy db`, or a direct database connection?

stevenGravy replied:
introduced in the beginning now

stevenGravy committed:
Include TLS routing info at top

stevenGravy commented on docs/pages/connect-your-client/gui-clients.mdx:
```diff
@@ -71,34 +71,16 @@ Ensure that your environment includes the following:
+
+(!docs/pages/includes/determinetls.mdx!)
+
+If TLS Routing is confirmed enabled use the following command to start a local TLS proxy your
```
should be uniform.

stevenGravy commented:
@ptgott please take a look at addressed comments

ptgott commented on docs/pages/connect-your-client/gui-clients.mdx:
```diff
@@ -69,36 +69,21 @@ Ensure that your environment includes the following:
 
 <ScopedBlock scope={["oss", "enterprise"]}>
```
Since users tend not to slick the scope switcher at the top of the page, we should use `Tabs` instead of a `ScopedBlock`.

ptgott commented on docs/pages/connect-your-client/gui-clients.mdx:
```diff
@@ -69,36 +69,21 @@ Ensure that your environment includes the following:
-Starting the local database proxy with the `--tunnel` flag will create an
-authenticated tunnel that you can use to connect to your database instances.
-You won't need to configure any credentials when connecting to this tunnel.
+The TLS Routing configuration on the Teleport Cluster impacts the available ports for
+database clients to connect.
```
```suggestion
The TLS Routing configuration on the Teleport cluster impacts the ports that are available for database clients to connect to.

Follow the instructions below to see if TLS Routing is enabled on your cluster and get connection information for your database GUI clients.
```

ptgott commented on docs/pages/connect-your-client/gui-clients.mdx:
```diff
@@ -69,36 +69,21 @@ Ensure that your environment includes the following:
-mode where each database protocol is multiplexed on the same web proxy port, use
-the following command to start a local TLS proxy your GUI database client will
-be connecting to:
+<TabItem label="TLS Routing enabled">
```
```suggestion
<TabItem label="TLS Routing Enabled">
```

ptgott commented on docs/pages/connect-your-client/gui-clients.mdx:
```diff
@@ -114,8 +99,10 @@ Use the displayed local proxy host/port and credentials paths when configuring
 </TabItem>
-<TabItem label="Separate ports">
-If you're not using TLS routing, run the following command to see the database
+<TabItem label="TLS Routing disabled">
```
```suggestion
<TabItem label="TLS Routing Disabled">
```

stevenGravy committed:
Apply suggestions from code review

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

stevenGravy committed:
Merge branch 'master' into stevenGravy/docs/tlsrouting

ptgott commented:
@stevenGravy Just checking whether you plan to finish incorporating the feedback on this PR

stevenGravy committed:
uses variable and rewords
